src/pg/configurator.rs

```diff
@@ -169,6 +221,41 @@ impl PgBuilder {
     }
 }
 
+fn new_auto_publish(config: &PgConfig, is_function: bool) -> Option<PgBuilderPublish> {
```

Comment:
I'm okay to go with the notion that auto publish is true by default, as mentioned here https://github.com/maplibre/martin/issues/512#issuecomment-1368259212
But it still looks confusing and hard to reason about for me. We'd need to make this behavior clear in the docs.